app/src/cose.h

```diff
@@ -700,10 +645,19 @@ namespace scitt::cose
     t_cose_sign1_verify_ctx verify_ctx;
 
     // Do some preliminary decoding, to get the header parameters and potential
-    // auxiliary buffer size.
+    // auxiliary buffer size. T_COSE_OPT_UNKNOWN_CRIT_ALLOWED
     t_cose_parameters params;
-    t_cose_sign1_verify_init(
-      &verify_ctx, T_COSE_OPT_TAG_REQUIRED | T_COSE_OPT_DECODE_ONLY);
+    uint32_t prelim_options;
+    if (allow_unknown_crit)
+    {
+      prelim_options = T_COSE_OPT_TAG_REQUIRED | T_COSE_OPT_DECODE_ONLY |
```

Comment:
I would avoid the duplication and in this line just do `prelim_options |= T_COSE_OPT_UNKNOWN_CRIT_ALLOWED;` The assignment in the else branch would be the initial assignment.